modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java

```diff
@@ -610,14 +618,53 @@ public boolean attemptReconnection()
         return false;
     }
 
-    public ProtocolVersion makeBrokerConnection(BrokerDetails brokerDetail) throws IOException, AMQException
+    public ProtocolVersion makeBrokerConnection(final BrokerDetails brokerDetail) throws IOException, AMQException
     {
-        return _delegate.makeBrokerConnection(brokerDetail);
+        // Requires permission java.lang.RuntimePermission "modifyThreadGroup"
+        Object returnObject = AccessController.doPrivileged(new PrivilegedAction() {
```

Comment:
Here the method returns as well as the method exceptions have been returned for propagation.